File: src/frontend/src/shared/ServerResponse.ts
import type { FullChat } from "./types";

export default interface ServerResponse<T> {
  status: Status;
  content: T;
}

export function generateResponse<T>(
  res: any,
  status: Status,
  content: T,
): ServerResponse<T> {
  return res.json({ status, content } as ServerResponse<T>);
}

export function sendResponse<T>(res: any, status: Status, content: T) {
  const response: ServerResponse<T> = {
    status,
    content,
  };
  res.json(response);
}

export interface Status {
  code: number;
  msg: string;
}

//ResponseTypes
export interface ChatPackage {
  username: string;
  fullChat: FullChat;
}

export interface ChatsPackage {
  username: string;
  chats: FullChat[];
}

export interface SilentResponse {
  emptyString: string;
}

export interface SearchMatch {
  username: string;
  color_id: number;
}

File: src/backend/src/utils/server_response.py
from fastapi.responses import JSONResponse
from src.utils.status import Status

def generate_response(status: Status, content: any = "") -> JSONResponse:
    return JSONResponse(
        status_code=200, 
        content={
            "status": {
                "code": status.value.code,
                "msg": status.value.msg
            },
            "content": content
        }
    )

def generate_ws_payload(status: Status, content: any = "") -> dict:
    return {
        "status": {
            "code": status.value.code,
            "msg": status.value.msg
        },
        "content": content
    }

def send_response(status: Status, content: any = "") -> JSONResponse:
    return generate_response(status, content)